application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,205 @@
+
+        if (handleIsOnCooldown(userID, event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+
+        String reportedMessage = event.getTarget().getContentRaw();
+        String reportedMessageUrl = event.getTarget().getJumpUrl();
+
+        TextInput body = TextInput
+            .create(REPORT_REASON_INPUT_ID, "Anonymous report to the moderators",
+                    TextInputStyle.PARAGRAPH)
+            .setPlaceholder("What is wrong with the message, why do you want to report it?")
+            .setRequiredRange(3, 200)
+            .build();
+
+        String componentID =
+                generateComponentId(reportedMessage, reportedMessageUrl, reportedMessageTimestamp);
+        Modal modal =
+                Modal.create(componentID, "Report this to a moderator").addActionRow(body).build();
+
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog = handleRequireModMailChannel(event, guildID);
+
+        if (modMailAuditLog.isEmpty()) {
+            return;
+        }
+
+        sendModMessage(event, args, modMailAuditLog.orElseThrow());
+    }
+
+    private Optional<TextChannel> handleRequireModMailChannel(ModalInteractionEvent event,
+            long guildID) {
+        Optional<TextChannel> modMailAuditLog = event.getJDA()
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+        if (modMailAuditLog.isEmpty()) {
+            event
+                .reply("Sorry, there was an issue sending your report to the moderators. We are "
+                        + "investigating.")
+                .queue();
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+        }
+        return modMailAuditLog;
+    }
+
+    private boolean handleIsOnCooldown(long userID, MessageContextInteractionEvent event) {
+        if (!isAuthorOnCooldown(userID)) {
+            return false;
+        }
+        event
+            .reply("You can only report a message once per %s minutes."
+                .formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isAuthorOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
```

Comment:
`handleIsOnCooldown` is just `isAuthorOnCooldown` but with a fancy message. You can get rid of `isAuthorOnCooldown` and put it into `handleIsOnCooldown`

Reply:
It looks better in a helper method, less crowded

Reply:
No you just created a useless method

Reply:
No